### PR TITLE
Add dummy scheduler in order to support building without nemo-keepali…

### PR DESCRIPTION
**Pull request by sunweaver**
…ve/libiphb.

This allows using buteo-syncfw on non-SFOS systems such as Ubuntu Touch or in other Linux distributions that ship the Lomiri Operating Environment.

### CODE REVIEW

**Comment by pvuorela** [comment]
The dummy scheduler doesn't seem to now add any functionality. Wondering would it be simpler if instead of having ifdef checks on USE_DUMMY_SCHEDULER the libiphb code would be opt-in with similar define like USE_KEEPALIVE.

**Comment by sunweaver** [comment]
> The dummy scheduler doesn't seem to now add any functionality. Wondering would it be simpler if instead of having ifdef checks on USE_DUMMY_SCHEDULER the libiphb code would be opt-in with similar define like USE_KEEPALIVE.

Sorry for the late response. Yes, this would be a good approach. Is this something you could work on? Or shall we provide a PR? (I sense I should not ask this question because I seem to know the answer... ;-) ).

**Comment by pvuorela** [comment]
> Sorry for the late response. Yes, this would be a good approach. Is this something you could work on? Or shall we provide a PR? (I sense I should not ask this question because I seem to know the answer... ;-) ).

I considered that to be evolution of this PR :)

**Comment by gber** [comment]
I have added a `USE_IPHB` macro.

File: msyncd/SyncScheduler.h [hunk 1]
```diff
@@ -128,7 +128,7 @@ private slots:
      */
     void doIPHeartbeatActions(QString aProfileName);
 
-#ifdef USE_KEEPALIVE
+#if defined(USE_IPHB)
```
Comment (pvuorela) [inline msyncd/SyncScheduler.h:131]: Should be keepalive here.

Comment (gber) [reply msyncd/SyncScheduler.h]: Indeed, fixed.

**Review by pvuorela — APPROVED** [review]

**Pull request merged**